lib/rhc/commands/port_forward.rb

```diff
@@ -97,9 +97,14 @@ def run(app)
             end
           end
         else
-          debug "Using #{ssh_executable} to determine forwarding ports."
-          ssh_cmd = "#{ssh_executable} #{ssh_uri.user}@#{ssh_uri.host} '#{list_ports_cmd} 2>&1'"
-          status, output = run_with_system_ssh(ssh_cmd)
+          debug "\nUsing #{ssh_executable} to determine forwarding ports."
+          ssh_cmd = "#{ssh_executable} #{ssh_uri.user}@#{ssh_uri.host} '#{list_ports_cmd}'"
+          begin
+            status, output = run_with_system_ssh(ssh_cmd)
+          rescue RHC::SSHCommandFailed
+            ex_forward_cmd = "#{ssh_executable} -N -L <source_port>:<gear_ip>:<destination_port>"
+            raise RHC::PortForwardFailedException.new("#{output + "\n" if options.debug}Error trying to collect ports to forward from app. You can try to collect the ports manually by running:\n  " + ssh_cmd + "\nThis command does not support the use of a custom ssh executable. You can try to foward the collected ports manually by running:\n  " + ex_forward_cmd)
```

Comment:
No, the 2>&1 inside quotes, applied to the `list_ports_cmd` is appropriate. ssh will return the STDOUT returned by the command, so it should be returning the error. Even using the below, the error is not reported:
```
ssh_cmd = "#{ssh_executable} #{ssh_uri.user}@#{ssh_uri.host} '#{list_ports_cmd} 2>&1' 2>&1"
```